File: p5_infinite_runner/states/jumping_state.js
class JumpingState {
  constructor(actor) {
    const jumpForce = createVector(0, -10);
    actor.applyForce(jumpForce);
  }

  handleInput(_command, _actor) {}

  update(actor) {
    if (actor.location.y + actor.height > height - gameState.getFloorHeight) {
      actor.location.y = height - actor.height - gameState.getFloorHeight;
      actor.velocity.y = 0;
      actor.state = standingState;
    } else {
      actor.applyForce(gameState.getGravity);
    }
  }
}

File: ggez_infinite_runner/src/states/jumping_state.rs
use super::{Commands, Player, State};

pub struct JumpingState {}

impl JumpingState {
    pub fn new() -> JumpingState {
        JumpingState {}
    }
}

impl State for JumpingState {
    fn handle_input(&self, _command: &Commands, player: &mut Player) -> Option<Box<dyn State>> {
        None
    }

    fn update(&self) {}
}
